Inclusion of the new splitting in Navier-Stokes embedded element

rubenzorrilla opened the pull request.

rubenzorrilla committed:
Header inclusion correction

rubenzorrilla committed:
Advances in new splitting addition.

The volume contribution as well as the new data structure have been updated. The boundary terms are still pending.

rubenzorrilla committed:
Preliminary working version. More checks are needed.

rubenzorrilla committed:
Clean up

Removing unused variables, adding some const statements and documentation.

rubenzorrilla committed:
Copy instead of reference in member variable

In order to avoid a memory leak, the nodal distances are now stored as a copy instead of storing them as a reference.

rubenzorrilla commented on kratos/modified_shape_functions/modified_shape_functions.h:
```diff
@@ -288,7 +288,7 @@ class KRATOS_API(KRATOS_CORE) ModifiedShapeFunctions
 
     const GeometryPointerType mpInputGeometry;
-    const Vector& mrNodalDistances;
+    const Vector mNodalDistances;
```
@RiccardoRossi, to avoid a memory leak when running a real case, I needed to store the nodal distances as a copy. This doesn't happen in the modified shape function tests in the core... Any idea?

RiccardoRossi replied:
a question: wouldn't it be enough to have a const reference to the distances?

rubenzorrilla replied:
It did not work... I don't now the reason...

rubenzorrilla committed:
Minor correction in splitting

Correction in both triangle and tretrahedron splitting algorithms. Besides, an extra test has been added to chech the subgeometries volumes and areas consistency.

rubenzorrilla committed:
Merged branch master into feature-new-splitting-embedded-elements

rubenzorrilla committed:
Extrapolation matrix and interface shape function gradients correction

rubenzorrilla committed:
Minor corrections and documentation update.

rubenzorrilla committed:
Addition of embedded 2D Couette tests

RiccardoRossi commented on applications/FluidDynamicsApplication/tests/EmbeddedCouette2DTest/EmbeddedCouette2DTest_parameters.json:
```diff
@@ -0,0 +1,131 @@
+        "kratos_module" : "KratosMultiphysics",
+        "process_name"  : "AssignScalarVariableProcess",
+        "Parameters"    : {
+            "mesh_id"         : 0,
```
mesh_id is not needed

rubenzorrilla committed:
LHS and RHS embedded element test

rubenzorrilla committed:
Fluid dynamics tests clean up

RiccardoRossi commented:
should this also go in the release? or is it the first time you merge this feature

rubenzorrilla commented:
Well, the new splitting and modified shape functions utilities are in the release branch (including the bug we solve today). However, these utilities are not used since the old version of the embedded element is the one included. Maybe it's a good idea to merge it (we would have the 2D).

RiccardoRossi commented:
i think that if it is a new feature it should not go in. I thought it was merged and broken. If this is the case i would at least push the bugfix
